crates/agentgateway/src/llm/vertex.rs

```diff
@@ -22,19 +26,69 @@ impl super::Provider for Provider {
 }
 
 impl Provider {
+	fn configured_model<'a>(&'a self, request_model: Option<&'a str>) -> Option<&'a str> {
+		self.model.as_deref().or(request_model)
+	}
+
+	fn anthropic_model<'a>(&'a self, request_model: Option<&'a str>) -> Option<Strng> {
+		let model = self.configured_model(request_model)?;
+		model
+			.strip_prefix("publishers/anthropic/models/")
+			.or_else(|| model.strip_prefix("anthropic/"))
+			.map(strng::new)
+	}
+
+	pub fn is_anthropic_model(&self, request_model: Option<&str>) -> bool {
+		self.anthropic_model(request_model).is_some()
+	}
+
+	pub fn prepare_anthropic_request_body(&self, body: Vec<u8>) -> Result<Vec<u8>, AIError> {
+		let mut map: Map<String, Value> =
```

Comment:
Can be a followup: instead of 2x marshal roundtrips, I would modify `to_anthropic` to take in an option to do this.